tools/ascli-actions-mdata.c

```diff
@@ -739,3 +739,69 @@ ascli_check_is_satisfied (const gchar *fname_or_cid, const gchar *cachepath, gbo
 
 	return res? ASCLI_EXIT_CODE_SUCCESS : ASCLI_EXIT_CODE_FAILED;
 }
+
+gint
+ascli_get_latest_version_file (const gchar *fname)
+{
+	g_autoptr(AsMetadata) metad = NULL;
+	g_autoptr(AsComponent) cpt = NULL;
+	g_autoptr(GFile) infile = NULL;
+	g_autoptr(GError) error = NULL;
+
+	if (fname == NULL) {
+		ascli_print_stderr (_("You need to specify an input file."));
+		return 1;
+	}
+
+	/* load input file */
+	infile = g_file_new_for_path (fname);
+	if (!g_file_query_exists (infile, NULL)) {
+		ascli_print_stderr (_("Metadata file '%s' does not exist."), fname);
+		return 2;
+	}
+
+	metad = as_metadata_new ();
+	as_metadata_set_locale (metad, "ALL");
+
+	if ((g_str_has_suffix (fname, ".yml.gz")) ||
+	    (g_str_has_suffix (fname, ".yaml.gz")) ||
+	    (g_str_has_suffix (fname, ".yml")) ||
+	    (g_str_has_suffix (fname, ".yaml"))) {
+		/* if we have YAML, we also automatically assume a catalog style */
+		as_metadata_set_format_style (metad, AS_FORMAT_STYLE_CATALOG);
```

Comment:
Does it make any sense to read this for `CATALOG` data? Since catalog files contain many components, you would currently just get a random version, so I don't think that makes sense.
I think it would be better if this code would just error-out in that case and limit itself to metainfo files.
The added benefit of that would be that this code would be much smaller then too ;-)